fastify.js

```diff
@@ -597,11 +610,19 @@ function build (options) {
       this._hooks.validate(name, fn)
       onRouteHooks.push(fn)
     } else {
-      this._hooks.add(name, fn.bind(this))
+      this.after((err, done) => {
+        _addHook(this, name, fn)
+        done(err)
+      })
     }
     return this
   }
 
+  function _addHook (instance, name, fn) {
+    instance._hooks.add(name, fn.bind(instance))
```

Comment:
while we are changing this, how about we move `_hooks` to `Symbol('hooks')` as well?

Reply:
I'm not sure about this. If we use a Symbol for `_hooks`, should we do the same for `_middlewares` as well? Should we change all `fastify._*` property keys to Symbols? Or is there a specific reason why some keys should be a Symbol and some should be an underscore `_*`?

Reply:
As long as all the property with Symbols are only used during the initial phase is ok. If we are using that property during the normal lifecycle of Fastify, I prefer we use classic objects keys, since they are slightly faster.

Reply:
Almost all of the `fastify._*` properties are only used during the initial startup phase, so should we use Symbols for all of them? I think keeping them as a regular string key is helpful for debugging: `console.log(fastify._hooks)`. But if we want to use Symbols, I think it would be better to do that in a separate PR. Changing `_hooks` here would create 14 lines of changes that are completely unrelated to the goal of this PR.

Reply:
👍 let's do another PR.